### PR TITLE
Adding a new pr template

### PR DESCRIPTION
**Pull request by anupambhatnagar**
Adding a template for the PR's to this repo.

### CODE REVIEW

**Comment by anupambhatnagar** [comment]
Canceled the Yamato tests manually as this will cause unnecessary load on the system.

**Comment by surfnerd** [comment]
looks like some whitespace failures for the `pre-commit` markdown checks.  LGTM otherwise

File: .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### Proposed change(s)
+
+Describe the changes made in this PR. If this is a bug fix or a new feature please provide a link
```
Comment (chriselion) [inline .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: "provide a link to it" - link to what?

Maybe add a link section with items for github issues, forum threads, and jira tickets.

File: .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- [ ] Bug fix
+- [ ] New feature
+- [ ] Code refactor
+- [ ] Breaking change
+- [ ] Documentation update
+- [ ] Other (please describe)
+
+### Checklist
+- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (harperj) [inline .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: I would almost rather just have a section of "How was this change tested?"

File: .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (anupambhatnagar) [reply .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: added a section.

File: .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (anupambhatnagar) [reply .github/PR_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: I suggest that we merge this template and then refine it further as we start using it.

**Pull request merged**